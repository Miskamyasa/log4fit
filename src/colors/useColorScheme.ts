import {useColorScheme as _useColorScheme} from "react-native"

import type {ColorSchemeName} from "./types"

export function useColorScheme(): ColorSchemeName {
    return _useColorScheme() ?? "dark"
}
