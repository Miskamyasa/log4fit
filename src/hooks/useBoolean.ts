import {useCallback, useState} from "react";


type ReturnObject = [boolean, () => void, () => void];

function useBoolean(
  initialState = false,
  callbackOnTrue?: () => void,
  callbackOnFalse?: () => void
): ReturnObject {
  const [isTrue, setBoolean] = useState(initialState);

  const setTrue = useCallback(() => {
    setBoolean(true);
    if (callbackOnTrue) {
      callbackOnTrue();
    }
  }, [callbackOnTrue]);

  const setFalse = useCallback(() => {
    setBoolean(false);
    if (callbackOnFalse) {
      callbackOnFalse();
    }
  }, [callbackOnFalse]);

  return [isTrue, setTrue, setFalse];
}

export default useBoolean;
