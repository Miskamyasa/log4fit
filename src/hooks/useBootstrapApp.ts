import {useEffect, useState} from "react"
import {Alert} from "react-native"

import NetInfo from "@react-native-community/netinfo"
import {Asset} from "expo-asset"
import * as SplashScreen from "expo-splash-screen"
import * as Updates from "expo-updates"

import {imagesToLoad} from "../../assets/images"
import {analytics} from "../helpers/analytics"
import {__t} from "../helpers/i18n"

async function loadResourcesAndDataAsync(onDone: () => void): Promise<void> {
    try {
        await Promise.all([
            Asset.loadAsync(imagesToLoad),
        ])
    }
    catch (e) {
        analytics.trackError(e)
    }
    finally {
        onDone()
    }
}

async function versionCheck(): Promise<void> {
    if (__DEV__) {
        return
    }
    // eslint-disable-next-line import/no-named-as-default-member
    const {isConnected} = await NetInfo.fetch()
    if (isConnected) {
        const {isAvailable} = await Updates.checkForUpdateAsync()
        if (isAvailable) {
            const update = await Updates.fetchUpdateAsync()
            if (update.isNew) {
                Alert.alert(
                    "",
                    __t("newVersion.modalText"),
                    [
                        {text: __t("continue")},
                        {text: __t("newVersion.update"), onPress: (): void => void Updates.reloadAsync()},
                    ],
                    {cancelable: false},
                )
            }
        }
    }
}

export function useBootstrapApp(): boolean {
    const [isLoadingComplete, setLoadingComplete] = useState(false)

    useEffect(() => {
        void SplashScreen.preventAutoHideAsync()
        void loadResourcesAndDataAsync(() => {
            setTimeout(() => void versionCheck(), 3000)
            void SplashScreen.hideAsync()
            setLoadingComplete(true)
        })
    }, [])

    return isLoadingComplete
}
