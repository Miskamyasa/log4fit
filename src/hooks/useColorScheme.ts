import {ColorSchemeName as SchemeName, useColorScheme as _useColorScheme} from "react-native";

export type ColorSchemeName = NonNullable<SchemeName>;

export function useColorScheme(): ColorSchemeName {
  return _useColorScheme() || "dark";
}
