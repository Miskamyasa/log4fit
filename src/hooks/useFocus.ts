import {useCallback, useState} from "react";


function useFocus(initialState = false): [boolean, () => void, () => void] {
  const [inFocus, setInFocus] = useState(initialState);

  const onFocus = useCallback(() => setInFocus(true), []);
  const onBlur = useCallback(() => setInFocus(false), []);

  return [inFocus, onFocus, onBlur];
}

export default useFocus;
