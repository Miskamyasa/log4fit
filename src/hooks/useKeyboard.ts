import {useCallback, useEffect, useState} from "react"
import {Keyboard, Platform} from "react-native"

const showEvent = Platform.OS === "android" ? "keyboardDidShow" : "keyboardWillShow"
const hideEvent = Platform.OS === "android" ? "keyboardDidHide" : "keyboardWillHide"

type ReturnObject = [
    boolean,
    () => void,
    () => false,
]

export function useKeyboard(): ReturnObject {
    const [visible, setVisible] = useState(false)

    useEffect(() => {
        const showListener = Keyboard.addListener(showEvent, () => {
            setVisible(true)
            return true
        })
        const hideListener = Keyboard.addListener(hideEvent, () => {
            setVisible(false)
            return true
        })
        return (): void => {
            showListener.remove()
            hideListener.remove()
        }
    }, [])

    const dismiss = useCallback(() => {
        Keyboard.dismiss()
    }, [])

    const handleUnhandledTouches = useCallback(() => {
        dismiss()
        return false
    }, [dismiss])

    return [visible, dismiss, handleUnhandledTouches]
}
