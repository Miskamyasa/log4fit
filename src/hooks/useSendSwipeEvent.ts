import {useCallback} from "react"
import type {NativeScrollEvent, NativeSyntheticEvent} from "react-native"

import {pick} from "lodash"

import {analytics} from "../helpers/analytics"

const keys = ["contentOffset", "contentSize", "layoutMeasurement"]

export function useSendSwipeEvent(name: string): (ev: NativeSyntheticEvent<NativeScrollEvent>) => void {
    const sendSwipeEvent = useCallback(({nativeEvent}: NativeSyntheticEvent<NativeScrollEvent>) => {
        analytics.sendEvent(name, pick(nativeEvent, keys))
    }, [name])

    return sendSwipeEvent
}
