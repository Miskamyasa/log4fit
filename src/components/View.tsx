import {ReactElement} from "react";
import {View as RNView, StyleSheet} from "react-native";
import {useThemeColor, ThemeProps} from "../hooks/useThemeColor";

type Props = RNView["props"] & ThemeProps;

export default function View(props: Props): ReactElement {
  const {style, light, dark, ...otherProps} = props;
  const backgroundColor = useThemeColor("background", {light, dark});

  return (
    <RNView
      style={StyleSheet.compose({backgroundColor}, style)}
      {...otherProps} />
  );
}
