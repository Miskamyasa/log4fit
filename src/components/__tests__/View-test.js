import renderer from "react-test-renderer";
import {Text} from "react-native";
import View from "../View";
import {useColorScheme} from "../../hooks/useColorScheme";

jest.mock("../../hooks/useColorScheme");

describe("components: View", () => {
  const DEFAULT = "light";
  useColorScheme.mockReturnValue(DEFAULT);

  const children = <Text>View</Text>;

  // const {result: colorScheme} = renderHook(() => useColorScheme());

  it("renders with default (light theme)", () => {
    const tree = renderer.create(<View>{children}</View>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with dark theme", () => {
    const DARK = "dark";
    useColorScheme.mockReturnValue(DARK);
    const tree = renderer.create(<View>{children}</View>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with custom background color #AAA", () => {
    const background = "#AAA";
    const props = {light: background, dark: background};
    const tree = renderer.create(<View {...props}>{children}</View>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with custom styles {padding: 12}", () => {
    const props = {style: {padding: 12}};
    const tree = renderer.create(<View {...props}>{children}</View>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
