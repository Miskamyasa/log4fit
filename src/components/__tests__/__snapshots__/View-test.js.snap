// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components: View renders with custom background color #AAA 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#AAA",
    }
  }
>
  <Text>
    View
  </Text>
</View>
`;

exports[`components: View renders with custom styles {padding: 12} 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#000",
      },
      Object {
        "padding": 12,
      },
    ]
  }
>
  <Text>
    View
  </Text>
</View>
`;

exports[`components: View renders with dark theme 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000",
    }
  }
>
  <Text>
    View
  </Text>
</View>
`;

exports[`components: View renders with default (light theme) 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fefefe",
    }
  }
>
  <Text>
    View
  </Text>
</View>
`;
